src/defu.ts

```diff
@@ -57,8 +56,7 @@ function _isPlainObject(value: unknown): boolean {
     (prototype === null ||
       prototype === Object.prototype ||
       Object.getPrototypeOf(prototype) === null) &&
-    !(Symbol.toStringTag in value) &&
-    !(Symbol.iterator in value)
+    (!(Symbol.toStringTag in value) || !(Symbol.iterator in value))
```

Comment:
I'm wondering if it is purely for Date merging fix, using `instance of Date` would be faster

Reply:
it isn't. See the linked issue. It also applies to `Module` (import via *)

Reply:
Oh, I see! With this PR want to merge Module instances!

Reply:
Checking, this logic currently basically bypasses entire L59 as `iterator` symbol does not exist in any of tested values (removing the line also all the tests pass). Wondering how we can cover it

Reply:
> Oh, I see! With this PR want to merge Module instances!

yes indeed 👍🏻

Reply:
>  Checking, this logic currently basically bypasses entire L59 as iterator symbol does not exist in any of tested values (removing the line also all the tests pass). Wondering how we can cover it

Oh, I see 🤔 That's a bit tricky. I also wonder how to cover it then...

But - do we need to? Dates are correctly replaced (not merged) like that too

Reply:
I have ported test suite of is-plain-object (see commits to main) and also merged in with this PR. Intrestingly we will be breaking two more tests at least (`Math` and native `arguments` will be wrongly detected as pure object)

Reply:
Updated with a logic that precisely works for `Module` to reduce (more!) regression chances